### PR TITLE
lock funds up per escrow hash

### PR DESCRIPTION
**Pull request by lalexgap**
This is a quick solution to #8 that allows funds to be locked up per `escrowHash`.  This lets Alice lock up funds for multiple transfers.

### CODE REVIEW

File: contracts/L2.sol [hunk 1]
```diff
@@ -20,12 +20,12 @@ struct EscrowEntry {
 }
 
 contract L2Contract {
-    /// A record of escrow funds indexed by sender.
-    mapping(address => EscrowEntry) escrowEntries;
+    /// A record of escrow funds indexed by sende then escrowHash.
```
Comment (kerzhner) [inline contracts/L2.sol:23]:
```suggestion
    /// A record of escrow funds indexed by sender then escrowHash.
```

File: contracts/L2.sol [hunk 2]
```diff
@@ -109,7 +110,8 @@ contract L2Contract {
         Signature calldata firstSignature,
         WithdrawalTicket calldata secondTicket,
         Signature calldata secondSignature,
-        bytes32 escrowSecret
+        bytes32 escrowSecret,
+        bytes32 escrowHash
```
Comment (kerzhner) [inline contracts/L2.sol:114]: Should we compute the hash from the secret?

**Review by kerzhner — APPROVED** [review]

File: contracts/L2.sol (hunk 2 above)
Comment (lalexgap) [reply contracts/L2.sol:114]: 🤦  yup!

Comment (lalexgap) [reply contracts/L2.sol]: fea07aa

**Pull request merged**